paddle/fluid/framework/tensor.h

```diff
@@ -37,23 +37,84 @@ class Tensor {
 #ifdef PADDLE_WITH_MKLDNN
 
  public:
-  inline mkldnn::memory::format format() const { return format_; }
+  // TODO(jczaja): This is depracted and will be removed
+  inline mkldnn::memory::format format() const {
+    if (layout_ == DataLayout::kMKLDNN) {
+      return static_cast<mkldnn::memory::format>(mem_pd_.desc().data.format);
+    } else {
+      return mkldnn::memory::format::format_undef;
+    }
+  }
+
+  // TODO(jczaja): This is depracted and will be removed
+  inline void set_format(
+      const mkldnn::memory::format format,
+      mkldnn::memory::data_type data_type = mkldnn::memory::f32) {
+    // create primitive descriptor based on format
+    std::vector<int> ltz = paddle::framework::vectorize2int(dims());
+    auto md = mkldnn::memory::desc({ltz}, data_type, format);
+    auto& pool = platform::DeviceContextPool::Instance();
+    auto place = paddle::platform::CPUPlace();
+    auto* dev_ctx =
+        dynamic_cast<platform::MKLDNNDeviceContext*>(pool.Get(place));
+    auto& cpu_engine = dev_ctx->GetEngine();
+    mem_pd_ = mkldnn::memory::primitive_desc(md, cpu_engine);
+  }
+
+  inline mkldnn::memory::primitive_desc MKLDNNMemPrimDesc() const {
+    return mem_pd_;
+  }
+
+  inline void setMKLDNNMemPrimDesc(mkldnn::memory::primitive_desc& mem_pd) {
+    // Internally MKL-DNN is just copying (increasing reference counter)
+    // to shared_ptr. So asignment should be quite cheap
+    mem_pd_ = mem_pd;
+  }
 
-  inline void set_format(const mkldnn::memory::format format) {
-    format_ = format;
+  void CreateMemoryPrimDescFromDims(
```

Comment:
1. It will be done
2. Yes. This is correct. data() / mutable_data() , returns pointer to allocation. Op has to ask separately for layout and/or Memory Primitive descriptor. Perhaps it would be good to  extend function data()/ mutable_data() so they can be aware of layouts and return converted data. Definitely having data() function that can return pointer to data in Paddle Layout , would help with implementing unit test so they can operate on block layout , and then we get result in paddle layout (NCHW or other) for comparison with reference. As paddlepaddle supports many devices (CPU,GPU,MKL-DNN...) I would like someone more familiar with other devices to trigger change in interface.